src/jace/tracing.py

```diff
@@ -77,14 +77,20 @@ def make_jaxpr(
         - Handle default arguments of `fun`.
         - Handle static arguments.
         - Turn `trace_options` into a `TypedDict` and sync with `jace.jit`.
+
+    Note:
+        Not all arguments that are supported by `jax.make_jaxpr()` are also supported
+        by this function. There is no error checking performed.
     """
-    if trace_options:
-        raise NotImplementedError(
-            f"Not supported tracing options: {', '.join(f'{k}' for k in trace_options)}"
-        )
     # TODO(phimuell): Test if this restriction is needed.
     assert all(param.default is param.empty for param in inspect.signature(fun).parameters.values())
 
+    # NOTE: `jax.make_jaxpr()` to current tracing backend we use, never supported
+    #   all arguments `jax.jit()` supported, which is strange. But in JaCe this
+    #   _will_ not be the case. To make things work, until we have a better working
+    #   backend we have to filter, i.e. clearing the `trace_options`.
```

Comment:
Check and rewrite, I don't understand it.

Reply:
You are right it was written very strangely.